### PR TITLE
Feat: creation of the mini yak!

### PR DESCRIPTION
**Pull request by sidduHERE**
image

### CODE REVIEW

File: contracts/core/MiniYak.sol [hunk 1]
```diff
@@ -2,9 +2,45 @@
 pragma solidity ^0.8.0;
 
 import "../libs/openzeppelin/token/ERC20/ERC20.sol";
+import "../libs/openzeppelin/utils/math/safeMath.sol";
```
Comment (ASlowCheetah) [inline contracts/core/MiniYak.sol]: Not necessary, this is solidity 0.8+

File: contracts/core/MiniYak.sol [hunk 2]
```diff
@@ -2,9 +2,45 @@
...
+
+    /**
+     * @notice unmoons mini Yak to Yak and burns mini Yak
+     * @param amount amount of miniyak 
+     * @param to address of caller or the address to which YAK would be transferred to
+     */
+    function unmoon(uint amount, address to) external {
+        uint burn_amount = Math.min(amount, this.balanceOf(msg.sender));
+        _burn(msg.sender, burn_amount);
+        require(IERC20(YAK).transfer(to, burn_amount));
```
Comment (ASlowCheetah) [inline contracts/core/MiniYak.sol]: perhaps we should use OpenZeppelin SafeERC's safeTransfer for those instead

**Review by ASlowCheetah — COMMENTED** [review]
On a high level everything looks fine, should we write a small test just to make sure we can "moon" the smallest unit of YakToken and we can "unmoon" the smallest unit of MiniYak, just as an edge check?

File: contracts/core/MiniYak.sol (hunk 2 above)
Comment (sidduHERE) [reply contracts/core/MiniYak.sol]: Updated both the methods

File: test/MiniYak.ts [hunk 3]
```diff
@@ -1,30 +1,138 @@
...
+
+    beforeEach(async function () {
         owner = (await ethers.getSigners())[0]
         account = (await ethers.getSigners())[1]
-    })
-
-    it("Can be deployed", async function() {
         const factory = await ethers.getContractFactory("MiniYak")
         miniYakContract = await factory.connect(owner).deploy()
         // await expect(miniYakContract.deployed()).to.be.reverted
```
Comment (ASlowCheetah) [inline test/MiniYak.ts]: let's remove this

File: test/MiniYak.ts [hunk 4]
```diff
@@ -1,30 +1,138 @@
...
         await miniYakContract.deployed()
+        await intial_data_setup().catch();
+        
+    })
+
+    // it("Can be deployed", async function() {
```
Comment (ASlowCheetah) [inline test/MiniYak.ts]: lets also delete these

File: test/MiniYak.ts [hunk 5]
```diff
@@ -1,30 +1,138 @@
...
+    //     const factory = await ethers.getContractFactory("MiniYak")
+    //     miniYakContract = await factory.connect(owner).deploy()
+    //     // await expect(miniYakContract.deployed()).to.be.reverted
+    //     await miniYakContract.deployed()
+    // })
+
+    it("Check for decimals", async function () {
+        expect(await miniYakContract.decimals()).to.eq(
```
Comment (ASlowCheetah) [inline test/MiniYak.ts]: unnecessary break line

**Review by ASlowCheetah — CHANGES_REQUESTED** [review]
lets gooooo, final small changes

**Comment by ASlowCheetah** [comment]
Excelent work!

**Review by ASlowCheetah — APPROVED** [review]

**Pull request merged**